src/extension/content-script/batteries/Reddit.ts

```diff
@@ -0,0 +1,76 @@
+import getOriginData from "../originData";
+import setLightningData from "../setLightningData";
+
+const urlMatcher = /^https:\/\/www.reddit\.com\/user\/(\w+).*/;
+
+// can we detect the user description from the description metatag?
+function isOnProfilePage() {
+  return (
+    document.querySelector<HTMLMetaElement>(
+      'head > meta[name="description"]'
+    ) != null
+  );
+}
+
+function getUserData() {
+  if (isOnProfilePage()) {
+    const element = document.querySelector<HTMLMetaElement>(
+      'head > meta[name="description"]'
+    );
+    const imageUrl = document.querySelector<HTMLImageElement>(
+      `img[alt="User avatar"]`
+    )?.src;
+    if (element && imageUrl) {
+      return {
+        element,
+        imageUrl,
+        name: document.title,
+      };
+    }
+  }
+  return null;
+}
+
+function battery(): void {
+  let userData;
+  if ((userData = getUserData())) {
+    let match;
+    let recipient;
+    // attempt to extract lnurlp: from the description text
+    if ((match = (userData.element.content || "").match(/lnurlp:(\S+)/i))) {
+      recipient = match[1];
+    } else {
+      // else let's look for an ⚡ emoji in the userDescription
+      if (userData.element.content && userData.element.content.includes("⚡")) {
+        const descriptionText = userData.element.content.split(" ");
```

Comment:
Twitter replaces the emoji with an image tag which is the reason why we can not use a regex. 
Reddit seems a bit nicer there and we should be able to use the regex.

so we should be able to do something like here: https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/content-script/batteries/YouTubeVideo.ts#L20-L29

```js
const text = descriptionElement.content;
// check for an lnurl
if ((match = text.match(/(lnurlp:)(\S+)/i))) {
  recipient = match[2];
}
// if there is no lnurl we check for a zap emoji with a lightning address
// we check for the @-sign to try to limit the possibility to match some invalid text (e.g. random emoji usage)
else if ((match = text.match(/(⚡:?|lightning:|lnurl:)\s?(\S+@\S+)/i))) {
  recipient = match[2];
} else {
  return;
}
```